operators/advanced_network/python/check_system.py

```diff
@@ -0,0 +1,632 @@
+
+        # Parse the output to extract interface names and PCIe addresses
+        names = []
+        addrs = []
+        for line in result.stdout.splitlines():
+            match = re.match(r"([\S\:\.]+) .*==>\s+(\S+)", line)
+            if match:
+                pcie_address = match.group(1)
+                interface_name = match.group(2)
+                names.append(interface_name)
+                addrs.append(pcie_address)
+
+        return names, addrs
+
+    except FileNotFoundError:
+        print(
+            "The ibdev2netdev command is not found. Ensure that it is installed and available in your PATH."
+        )
+        return [], []
+    except subprocess.CalledProcessError as e:
+        print(f"Error while executing ibdev2netdev: {e}")
+        return [], []
+    except Exception as e:
+        print(f"An unexpected error occurred: {e}")
+        return [], []
+
+
+def get_online_cpus():
+    """
+    Returns a list of online CPUs by reading /sys/devices/system/cpu/online.
+    """
+    try:
+        with open("/sys/devices/system/cpu/online", "r") as f:
+            online_cpus = f.read().strip()
+
+        # Parse ranges (e.g., "0-3" -> [0, 1, 2, 3])
+        cpu_list = []
+        for part in online_cpus.split(","):
+            if "-" in part:
+                start, end = map(int, part.split("-"))
+                cpu_list.extend(range(start, end + 1))
+            else:
+                cpu_list.append(int(part))
+
+        return cpu_list
+    except FileNotFoundError:
+        logging.error(
+            "Could not determine online CPUs. File /sys/devices/system/cpu/online not found."
+        )
+        sys.exit(1)
+
+
+def check_cpu_governor():
+    """
+    Checks if the CPU frequency governor is set to 'performance' for all online CPUs.
+    """
+    online_cpus = get_online_cpus()
+
+    for cpu in online_cpus:
+        scaling_governor_path = f"/sys/devices/system/cpu/cpu{cpu}/cpufreq/scaling_governor"
+
+        try:
+            with open(scaling_governor_path, "r") as f:
+                governor = f.read().strip()
+
+            if governor == "performance":
+                logging.info(f"CPU {cpu}: Governor is correctly set to 'performance'.")
+            else:
+                logging.warning(f"CPU {cpu}: Governor is set to '{governor}', not 'performance'.")
+
+        except FileNotFoundError:
+            logging.error(
+                f"CPU {cpu}: Scaling governor file not found. This CPU may not support frequency scaling."
+            )
+        except PermissionError:
+            logging.error(
+                f"CPU {cpu}: Permission denied while accessing scaling governor file. Run as root."
+            )
+
+
+def check_mrrs():
+    """
+    Checks if the Maximum Read Request Size (MRRS) of NVIDIA Ethernet controllers
+    is set to 4096.
+    """
+    try:
+        _, addrs = get_nic_info()
+
+        for pci_address in addrs:
```

Comment:
please loop over interface name as well so we can print it in the log

Example:
```log
2024-12-09 10:39:20 - WARNING - 0005:03:00.0: MRRS is set to 4194304, not 4096.
```

Would be

```log
2024-12-09 10:39:20 - WARNING - NIC enP5p3s0f0np0 [0005:03:00.0]: MRRS is set to 4194304, not 4096.
```